core-bundle/src/Command/ResizeImagesCommand.php

```diff
@@ -85,181 +102,250 @@ protected function configure(): void
-            ->addOption('throttle', 't', InputOption::VALUE_OPTIONAL, 'Pause between resizes to limit CPU utilization, 0.1 relates to 10% CPU usage', '1')
+            ->addOption('concurrent', 'c', InputOption::VALUE_OPTIONAL, 'Run multiple processes concurrently with a value larger than 1 or pause between resizes to limit CPU utilization with values lower than 1.0', '1')
+            ->addOption('throttle', 't', InputOption::VALUE_OPTIONAL, '(Deprecated) Use the concurrent option instead', '1')
+            ->addOption('image', null, InputArgument::OPTIONAL, 'Image name to resize a single image')
             ->setDescription('Resizes deferred images that have not been processed yet.')
         ;
     }
 
     protected function execute(InputInterface $input, OutputInterface $output): int
     {
+        $this->io = new SymfonyStyle($input, $output->section());
+
         if (!$this->resizer instanceof DeferredResizerInterface) {
             throw new \RuntimeException('Deferred resizer not available');
         }
 
-        $timeLimit = (float) $input->getOption('time-limit');
-        $throttle = (float) $input->getOption('throttle');
-        $concurrent = (int) $input->getOption('concurrent');
-
-        if ($timeLimit < 0) {
-            throw new InvalidArgumentException(sprintf('Time-limit value "%s" is invalid.', $timeLimit));
-        }
-
-        if ($throttle < 0.001 || $throttle > 1) {
-            throw new InvalidArgumentException(sprintf('Throttle value "%s" is invalid.', $throttle));
-        }
-
-        if ($concurrent < 1) {
-            throw new InvalidArgumentException(sprintf('Concurrent value "%s" is invalid.', $concurrent));
+        if (!$output instanceof ConsoleOutputInterface) {
+            throw new \InvalidArgumentException(sprintf('Output must be an instance of "%s"', ConsoleOutputInterface::class));
         }
 
-        $io = new SymfonyStyle($input, $output);
-
-        if ($concurrent > 1) {
-            return $this->executeConcurrent($concurrent, $io);
+        if (null !== $image = $input->getOption('image')) {
+            return $this->resizeImage($image);
         }
 
-        $startTime = microtime(true);
-
-        $maxDuration = 0;
-        $lastDuration = 0;
-
-        foreach ($this->storage->listPaths() as $path) {
-            $sleep = (1 - $throttle) / $throttle * $lastDuration;
+        $timeLimit = (float) $input->getOption('time-limit');
+        $throttle = (float) $input->getOption('throttle');
+        $concurrent = (float) $input->getOption('concurrent');
 
-            if ($timeLimit && microtime(true) - $startTime + $maxDuration + $sleep > $timeLimit) {
-                $io->writeln("\n".'Time limit of '.$timeLimit.' seconds reached.');
+        if (1.0 !== $throttle) {
+            @trigger_error('Using the throttle option is deprecated and will no longer work in Contao 5.0. Use the concurrent option instead.', E_USER_DEPRECATED);
```

Comment:
This would need to be adjusted when merging upstream to 4.10 or master @leofeyer. I'm not sure when the switch to `trigger_deprecation()` came.

Reply:
Yes, 4.10.

Reply:
See #2041.